### ISSUE TITLE
aws.KinesisVideo().getMedia() won't trigger the callback

### ISSUE THREAD

**Issue by davidgatti** [aws/aws-sdk-js#2013]
Hello, 

I have the following situation where the callback of the `.getMedia()` function is never triggered despite data coming in, as shown in the screenshot bellow.

![screen shot 2018-04-12 at 2 42 41 pm](https://user-images.githubusercontent.com/411114/38934730-bd6e0c92-431c-11e8-9fac-1794ebe3a5e9.png)

And bellow there is another image showing that the NodeJS process is getting fatter and fatter ;) 

![screen shot 2018-04-12 at 2 47 49 pm](https://user-images.githubusercontent.com/411114/38934753-c8e2df62-431c-11e8-83b7-27ff6a49966d.png)

And that is it. I can't console log anything since the callback is never executed by your SDK.

Also in your documentation (screenshot added) you have a sentence that stops in the middle.

![screen shot 2018-04-12 at 2 57 01 pm](https://user-images.githubusercontent.com/411114/38934791-e6561622-431c-11e8-9a79-78e5daa83a53.png)

I've attached the code bellow for reference.

``` javascript
let aws = require('aws-sdk')
let fs = require('fs');

//
//	1.	Create KV
//
let kinesisvideo = new aws.KinesisVideo({
	apiVersion: '2017-09-30',
	region: 'us-east-1',
	accessKeyId: 'SECRET',
	secretAccessKey: 'SECRET'
});

//
//	1. Prepare the parameters for KinesisVideo
//
let params = {
	APIName: 'GET_MEDIA',
	StreamARN: 'arn:aws:kinesisvideo:us-east-1:SECRET:stream/staging/SECRET'
};

//
//	3.	Execute the query
//
kinesisvideo.getDataEndpoint(params, function(error, data) {

	//
	//	1.	Check for errors
	//
	if(error)
	{
		console.log(error);
	}

	//	2.	Create Media Store
	//
	let kinesisvideomedia = new aws.KinesisVideoMedia({
		apiVersion: '2017-09-30',
		region: 'us-east-1',
		accessKeyId: 'SECRET',
		secretAccessKey: 'SECRET',
		endpoint: data.DataEndpoint
	});

	//
	//	3.	Create the parameters for getMedia()
	//
	let options = {
		StartSelector: {
			StartSelectorType: 'EARLIEST'
		},
		StreamARN: 'arn:aws:kinesisvideo:us-east-1:SECRET:stream/staging/SECRET'
	};

	//
	//	3.	Execute the query
	//
	kinesisvideomedia.getMedia(options, function(error, data) {

		//
		//	1.	Check for errors
		//
		if(error)
		{
			console.log(error);
		}

		//
		//	->	Show the bits that we should get
		//
		console.log(data);

	});

});

**Comment by davidgatti** [comment]
Hi @chrisradek,

About your question:

1. after 5 min I decided to stop 
2. I did try your suggestion and it worked, and it is actually what I was looking for.

Additional question to you:

1. How come createReadStream() information is missing from the documentation? At least I did not see any indication that I can create a Read Stream. After reading the documentation I assumed that you will call the callback at every frame. 
2. Eventually I'm going to stream the video to the browser but just for now, how can I save the video to disk? If I make a file called `movie.mkv` I just see the initial key frame, and done. Despite the file being 100mb
